### ISSUE TITLE
Extra single quote appeared after use safeDump

### ISSUE THREAD

**Issue by oMaten** [nodeca/js-yaml#349]
I use `safeDump` method to create a YAML document and here is my code

```js

const yamlContent = yaml.safeDump({
  build: [
    'npm -v && npm install',
    'npm run build -- --json --verbose --outputPath ./dist --config ./webpack.config.js --no-compress ' // here is a white space in last
  ]
}, { lineWidth: 120 });

console.log(yamlContent);
// build:
//  - npm -v && npm install
//  - 'npm run build -- --json --verbose --outputPath ./dist --config ./webpack.config.js --no-compress '
```

then i find single quote disappeared after i trim my string in array

```js

const yamlContent = yaml.safeDump({
  build: [
    'npm -v && npm install',
    'npm run build -- --json --verbose --outputPath ./dist --config ./webpack.config.js --no-compress '.trim()
  ]
}, { lineWidth: 120 });

console.log(yamlContent);
// build:
//  - npm -v && npm install
//  - npm run build -- --json --verbose --outputPath ./dist --config ./webpack.config.js --no-compress
```

**Comment by dgendill** [comment]
I ran into a similar situation, but with an object instead of an array.

I'm not familiar enough with YAML to know how older version of yaml worked, but it looks like the extra quotes get added in "compatability mode".  When I set `noCompatMode` to true when dumping, I no longer get the extra quotes.

Relevant code seems to be in lib/js-yaml/dumper on line 323.  This might be related to what you're seeing.
